index.js

```diff
@@ -187,177 +387,206 @@ function mergeResults(contentArray) {
+      } else {
+        debug = "false";
+      }
+
+      if (failBuild.toLowerCase() === "true") {
+        failBuild = true;
+      } else {
+        failBuild = false;
+      }
+
+      if (acsReportEnable.toLowerCase() === "true") {
+        acsReportEnable = true;
+      } else {
+        acsReportEnable = false;
+      }
+
+      if (
+        !SEVERITY_LIST.some(
+          (item) =>
+            typeof severityCutoff === "string" && item === severityCutoff
+        )
+      ) {
+        throw new Error(
+          `Invalid severity-cutoff value is set ${debug} ${severityCutoff} - please ensure you are choosing either Negligible, Low, Medium, High, or Critical`
+        );
+      }
+
+      if (!version) {
+        version = `${grypeVersion}`;
+      }
+
+      core.debug(`Installing grype version ${version}`);
+      await installGrype(version);
+
+      core.debug("Image: " + source);
+      core.debug("Debug Output: " + debug);
+      core.debug("Fail Build: " + failBuild);
+      core.debug("Severity Cutoff: " + severityCutoff);
+      core.debug("ACS Enable: " + acsReportEnable);
+
+      core.debug("Creating options for GRYPE analyzer");
+
+      // Run the grype analyzer
+      let cmdOutput = "";
+      let stdErr = "";
+      let cmd = `${grypeBinary}`;
+      let cmdArgs = [`-vv`, `-o`, `json`];
+      if (severityCutoff != "") {
+        cmdArgs.push("--fail-on");
+        cmdArgs.push(severityCutoff.toLowerCase());
+      }
+      cmdArgs.push(source);
+      const cmdOpts = {};
+      cmdOpts.listeners = {
+        stdout: (data = Buffer) => {
+          cmdOutput += data.toString();
+        },
+        stderr: (data = Buffer) => {
+          stdErr += data.toString();
+        },
+      };
 
-        core.debug('Image: ' + imageReference);
-        core.debug('Dockerfile path: ' + dockerfilePath);
-        core.debug('Inline Scan Image: ' + inlineScanImage);
-        core.debug('Debug Output: ' + debug);
-        core.debug('Fail Build: ' + failBuild);
-        core.debug('Include App Packages: ' + includePackages);
-        core.debug('Custom Policy Path: ' + customPolicyPath);
-        core.debug('ACS Enable: ' + acsReportEnable);	
-        core.debug('ACS Severity Cutoff: ' + acsSevCutoff);	
+      cmdOpts.ignoreReturnCode = true;
 
-        core.debug('Policy path for evaluation: ' + policyBundlePath);
-        core.debug('Policy name for evaluation: ' + policyBundleName);
+      core.info("\nAnalyzing: " + source);
+      core.debug(`Running cmd: ${cmd} ` + cmdArgs.join(" "));
+      let exitCode = await exec(cmd, cmdArgs, cmdOpts);
 
-        let cmd = `${runScan} ${scanScript} ${inlineScanImage} ${imageReference} ${debug} ${policyBundlePath} ${policyBundleName}`;
-        if (dockerfilePath) {
-            cmd = `${cmd} ${dockerfilePath}`
-        }
-        core.info('\nAnalyzing image: ' + imageReference);
-        await exec(cmd);
+      core.info("\nCaptured stderr from grype:\n" + stdErr);
```

Comment:
Looking at some of the pipeline runs, I'm seeing duplicate output due to the above line:
https://github.com/anchore/scan-action/runs/1178746750?check_suite_focus=true#step:3:48
https://github.com/anchore/scan-action/runs/1178746750?check_suite_focus=true#step:3:137

We should probably not capture stderr since it is already streamed to the log and we are not doing anything additional with that output other than printing it again.

Reply:
Good catch! I'll update this to remove this line.

Reply:
This has been removed!